src/backend/dx12/src/device.rs

```diff
@@ -2580,74 +2645,102 @@ impl d::Device<B> for Device {
                                 CounterOffsetInBytes: 0,
                                 Flags: d3d12::D3D12_BUFFER_UAV_FLAG_RAW,
                             };
+                            if heap.is_full() {
```

Comment:
why is this one here?

Reply:
I think it's good to be here. We make sure the pool index is pointing to a pool that can allocate for the next iteration.